src/passes/DeadArgumentElimination.cpp

```diff
@@ -693,7 +705,12 @@ struct DAE : public Pass {
         }
       }
     }
-    assert(lub.get() != originalType);
+
+    // We can compute the optimal LUB, which is hopefully not the same as the
+    // original type.
+    if (lub.get() == originalType) {
+      return false;
```

Comment:
This looks dangerous because `get()` could update nulls even if it returns the original type, in which case we would have to return `true` to refinalize. However, I think that if we get here, it must be that the lub is not `originalType` because we would have returned already after one of the calls to `processReturnValue` or `processReturnType`. So can we turn this into an assertion that the lub is not equal to `originalType`?

Reply:
Hmm, good point, this does look dangerous. I'll try to figure out something.

We can't assert here, because while you are right that the LUB is not `originalType`, it might still become `originalType` during the final computation - when we take into account nulls. But, that case is not dangerous to us here. It's annoying that the invariants that makes this safe are in the other file, which makes it hard to reason about safety here, which I'll think on.